pkg/controller/catalog/storage/mem_storage_instance.go

```diff
@@ -0,0 +1,63 @@
+package storage
+
+import (
+	"errors"
+
+	"github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog"
+)
+
+var (
+	errInstanceAlreadyExists = errors.New("instance already exists")
+	errNoSuchInstance        = errors.New("no such instance")
+)
+
+type memStorageInstance struct {
+	// maps instance ID to instance
+	instances map[string]*servicecatalog.Instance
+}
+
+func newMemStorageInstance() *memStorageInstance {
+	return &memStorageInstance{instances: make(map[string]*servicecatalog.Instance)}
+}
+
+func (m *memStorageInstance) List() ([]*servicecatalog.Instance, error) {
+	ret := make([]*servicecatalog.Instance, len(m.instances))
+	i := 0
+	for _, val := range m.instances {
+		ret[i] = val
+		i++
+	}
+	return ret, nil
+}
+
+func (m *memStorageInstance) Get(name string) (*servicecatalog.Instance, error) {
+	ret, ok := m.instances[name]
+	if !ok {
+		return nil, errNoSuchInstance
+	}
+	return ret, nil
```

Comment:
We may as well get this right, but not in this PR. Let's do it as part of the [concurrency safety PR I suggested](https://github.com/kubernetes-incubator/service-catalog/pull/176/files/5281cd13ec5b275f85ddb2c13e661cb7497fc3af#r93660942)